src/components/shared/Button.jsx

```diff
@@ -0,0 +1,72 @@
+import styled, { css } from "styled-components";
+
+const BUTTON_SIZES = {
+  md: css`
+    --button-min-width: 240px;
+    --button-min-height: 64px;
+    --button-font-size: 1rem;
+    --button-padding: 12px 24px;
+    --button-border: 2px solid #000;
+    --button-radius: 4px;
+  `,
+  lg: css`
+    --button-min-width: 420px;
+    --button-min-height: 64px;
+    --button-font-size: 1rem;
+    --button-padding: 12px 24px;
+    --button-border: 4px solid #000;
+    --button-radius: 4px;
+  `,
+};
+
+const USING_CASES = {
+  next: css`
+    --button-color: #fff;
+    --button-background-color: #000;
+  `,
+  before: css`
+    --button-color: #000;
+    --button-background-color: #000;
+  `,
```

Comment:
P3: black, white로 이루어진 라인/면 형태의 버튼은 이전, 다음 버튼 외로도 "새 프로젝트 비교하기", "현재 화면으로 이동하기(figma)", "취소하기", "비교하기" 버튼도 동일하게 사용되고 있습니다. 

아래같이 범용적인 이름을 쓰는 건 어떨 지 네이밍 건의 드립니다 :)

```suggestion
  solid: css`
    --button-color: #fff;
    --button-background-color: #000;
  `,
  line: css`
    --button-color: #000;
    --button-background-color: #000;
  `,
```

Reply:
혹시 solid와 line이 어떤 점에서 범용적인 의미라고 생각하셨는지 궁금합니다!

Reply:
버튼 형태로 생각해서 작성해봤습니다 :) 
이전/다음 버튼 외로도 범용적으로 쓰이게 될 속성인 것 같아서요.

예를 들어, 해당 버튼 속성으로 사용하게 될 취소/비교버튼에 next/before라는 명칭은 안 어울리는 듯해
solid, line이 아니더라도 버튼 명이 생각나는 속성명보다는 조금 더 일반적인 이름이 좋을 것 같아 
제안했었습니다 👀

Reply:
P5로 반영 안하셔도 좋을 것 같습니다!! 💯  이전/다음 의미도 좋네요!

Reply:
저도 저부분 네이밍을 할때 고민을 많이 했는데
슬기님이 어떤 생각의 흐름(?)으로 solid와 line이라는 변수명을 생각해 내신것 인지 궁금했습니다!

같이 공유하게 되면 다음에 네이밍을 짓는 경우 고민을 할 때 도움이 되지 않을까 해서요!

저같은 경우에는
저희 웹앱의 플로우 상
• 검정배경에 흰색을 한 버튼들은 `프로세스 흐름을 정방향`으로 흐르게 하는 버튼들이라고 생각해서 `다음`
• 나머지 버튼들은 `프로세스 흐름을 역방`향으로 흐르게 하는 버튼들이라고 생각해서 `이전`
이라는 네이밍을 지어봤습니다!

Reply:
성호님은 버튼을 누를 시 페이지 이동에 대한 주요 흐름(?)의 기능으로 구분하신 듯하고, 
저는 버튼의 디자인 UI 형태로만 구분을 해서 지으려했습니다. 
아래 글에 나온 solid/line button 형식 보시면 바로 이해되실 것 같습니다 :)

https://note-by-a.tistory.com/2